allneeds.lua

```diff
@@ -1,20 +1,110 @@
 -- Prints the sum of all citizens' needs.
+-- show only needs that needs attention now (damaging focus) and garantee it shows on dwarf overview (example need drink)
+-- with arg "n" it adds a TAG NDS_ NEEDS_CODE _NDS to citizens prefix names (example: allneeds n)
+-- with arg "r" it removes all NDS TAGS citizens prefix names (example: allneeds r)
+
+
+--SET SCOPE to handle unique elements (needs)
+local setid = {}
+local function addToSet(set, key)
+    set[key] = true
+end
+
+local function removeFromSet(set, key)
+    set[key] = nil
+end
+
+local function setContains(set, key)
+    return set[key] ~= nil
+end
+--SET SCOPE END
+
+--get only user custom nickname OR default
+local function getNick(unit)
+    local completeName = dfhack.TranslateName(dfhack.units.getVisibleName(unit))
+    local unit_name=string.sub(completeName,0,(string.find(completeName,"\' ") or string.len(completeName)))
+    unit_name=string.gsub(unit_name,"\'","")
+    unit_name=string.gsub(unit_name,"`","")
+    return unit_name
+end
+
+local function parse_commandline(args)
+    local opts = {}
+
+    for i,v in ipairs(args) do
+        if v == 'n' then opts.setnick = true return opts end
+        if v == 'r' then opts.removenick = true return opts end
+        if v == 'u' then opts.removenick = true opts.setnick = true return opts end
+    end
+
+    return opts
+end
+
+local opts = parse_commandline({...})
+
+local convertNeeds = {
+    ["Socialize"]                  = "S"    ,
+    ["DrinkAlcohol"]              = "DA"  ,
+    ["PrayOrMeditate"]              = "PM"  ,
+    ["StayOccupied"]              = "SO"  ,
+    ["BeCreative"]                  = "BC"   ,
+    ["Excitement"]                  = "E"   ,
+    ["LearnSomething"]              = "LS"  ,
+    ["BeWithFamily"]              = "Fa" ,
+    ["BeWithFriends"]              = "Fr"  ,
+    ["HearEloquence"]              = "H"     ,
+    ["UpholdTradition"]          = "T"     ,
+    ["SelfExamination"]          = "SE"     ,
+    ["MakeMerry"]                  =  "M"     ,
+    ["CraftObject"]              = "CO"     ,
+    ["MartialTraining"]          = "MT"    ,
+    ["PracticeSkill"]              = "PS"    ,
+    ["TakeItEasy"]                  =  "TE"     ,
+    ["MakeRomance"]              =  "MS"    ,
+    ["SeeAnimal"]                  =   "SA"    ,
+    ["SeeGreatBeast"]              =   "SGB"   ,
+    ["AcquireObject"]              =   "AO"   ,
+    ["EatGoodMeal"]              =   "EM"   ,
+    ["Fight"]                      =     "Fi"  ,
+    ["CauseTrouble"]              =    "CT"  ,
+    ["Argue"]                      =     "A"  ,
+    ["BeExtravagant"]              =     "BE" ,
+    ["Wander"]                     =      "W" ,
+    ["HelpSomebody"]              =     "HS" ,
+    ["ThinkAbstractly"]          =     "TA" ,
+    ["AdmireArt"]                  =     "AA" ,
```

Comment:
This table is difficult to read with the random extra whitespace. Could you reformat it?

Reply:
done